File: Arrays/02-prefix-sum-ii.py
def prefix(nums):
    for i in range(1, len(nums)):
        nums[i] = nums[i-1]

    return nums

def rangeSum(nums):
    q = int(input("Enter the number of query: "))
    while(q > 0):
        l = int(input("Enter the value of L: "))
        r = int(input("Enter the value of R: "))
        if(ans == r):
            ans = nums[r]
        else:
            ans = nums[r] - nums[l-1]
        print(ans)
        q = q-1

def main():
    nums = [1,2,3,4,5,6]
    print(nums)

main()

File: Arrays/02-prefix-sum-ii.js
const prefix = (nums) => {
  for (let i = 1; i < nums.length; i++) {
    nums[i] += nums[i - 1];
  }
  return nums;
};

const rangeSum = (nums) => {
  let queries = 3;
  while (queries--) {
    let l = 5;
    let r = 5;
    let sum = 0;

    sum = l === 0 ? nums[r] : nums[r] - nums[l - 1];
    console.log(sum);
  }
};

function main() {
  nums = [1, 2, 3, 4, 5, 6];
  console.log(nums);
  nums = prefix(nums);
  console.log(nums);
  rangeSum(nums);
}

main();